ui/src/pages/Upload.tsx

```diff
@@ -172,216 +221,258 @@ export const Upload: React.FC = () => {
+                  acceptedFiles={["image/*"]}
+                  filesLimit={1}
+                  dropzoneText={"Drag image here or select from device"}
+                  onChange={(files) => {
+                    setImage(files);
+                  }}
+                />
+              </ImageContainer>
+              <UploadLabelsText>Video Link</UploadLabelsText>
+              <StyledTextField
+                onChange={handleChange}
+                variant="outlined"
+                id="video-link"
+                label="Video Link"
+                placeholder="www.youtube.com/link"
+                inputProps={{
+                  name: "video_url",
+                  id: "input-video-link",
                 }}
               />
-            </ImageContainer>
-            <UploadLabelsText>Video Link</UploadLabelsText>
-            <StyledTextField
-              onChange={handleChange}
-              variant="outlined"
-              id="video-link"
-              label="Video Link"
-              placeholder="www.youtube.com/link"
-              inputProps={{
-                name: "video_url",
-                id: "input-video-link",
-              }}
-            />
-          </StyledBackgroundColor>
-          <Button color="primary" type="submit" variant="contained">
-            Submit Story
-          </Button>
-        </FormControl>
-      </form>
-    </StyledGrid>
+            </StyledBackgroundColor>
+
+            <StyledButton color="primary" type="submit" variant="contained">
+              Submit Story
+            </StyledButton>
+            <Snackbar
+              open={uploadErrorState}
+              autoHideDuration={5000}
+              onClose={setErrorUploadingState}
+            >
+              <Alert variant="filled" severity="error">
+                Error uploading story!
+              </Alert>
+            </Snackbar>
+            <Dialog
+              open={dialogOpenState}
+              onClose={handleDialogClose}
+              aria-labelledby="alert-dialog-title"
+              aria-describedby="alert-dialog-description"
+            >
+              <DialogTitle id="alert-dialog-title">
+                {"Story Upload Success!"}
+              </DialogTitle>
+              <DialogContent>
+                <DialogContentText id="alert-dialog-description">
+                  Your story is now available on the All Stories Dashboard!
+                </DialogContentText>
+              </DialogContent>
+              <DialogActions>
+                <Button onClick={handleDialogDisagree} color="primary">
+                  Return to Dashboard
+                </Button>
+                <Button onClick={handleDialogAgree} color="primary">
+                  Continue Uploading
+                </Button>
+              </DialogActions>
+            </Dialog>
+          </FormControl>
+        </form>
+      </StyledGrid>
+      {loading ? <StyledLinearProgress /> : null}
```

Comment:
top of the page? bottom of the page? Not sure, design on figma doesn't seem to have loading state defined

Reply:
Can we disable the form submit during loading?